tests/Xunit/J2N.Collections.Generic.List.Tests/List.Generic.Tests.EnsureCapacity.cs

```diff
@@ -0,0 +1,120 @@
+            Assert.Equal(0, list.Capacity);
+        }
+
+        [Fact]
+        public void EnsureCapacity_NegativeCapacityRequested_Throws()
+        {
+            var list = new JCG.List<T>();
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("capacity", () => list.EnsureCapacity(-1));
+        }
+
+        public static IEnumerable<object[]> EnsureCapacity_LargeCapacity_Throws_MemberData()
+        {
+            // J2N NOTE: This does not throw OutOfMemoryException in all platforms, and is not really a useful test in practice.
+            // yield return new object[] { 5, ArrayExtensions.MaxLength + 1 };
+            yield return new object[] { 1, int.MaxValue };
+        }
+
+        [Theory]
+        [MemberData(nameof(EnsureCapacity_LargeCapacity_Throws_MemberData))]
+        [InlineData(1, int.MaxValue)]
+        //[ActiveIssue("https://github.com/dotnet/runtime/issues/51411", TestRuntimes.Mono)]
+        public void EnsureCapacity_LargeCapacity_Throws(int count, int requestCapacity)
+        {
+            JCG.List<T> list = GenericListFactory(count);
+            Assert.Throws<OutOfMemoryException>(() => list.EnsureCapacity(requestCapacity));
+        }
+
+        [Theory]
+        [InlineData(5)]
+        public void EnsureCapacity_RequestedCapacitySmallerThanOrEqualToCurrent_CapacityUnchanged(int currentCapacity)
+        {
+            var list = new JCG.List<T>(currentCapacity);
+
+            for (int requestCapacity = 0; requestCapacity <= currentCapacity; requestCapacity++)
+            {
+                Assert.Equal(currentCapacity, list.EnsureCapacity(requestCapacity));
+                Assert.Equal(currentCapacity, list.Capacity);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void EnsureCapacity_RequestedCapacitySmallerThanOrEqualToCount_CapacityUnchanged(int count)
+        {
+            JCG.List<T> list = GenericListFactory(count);
+            var currentCapacity = list.Capacity;
+
+            for (int requestCapacity = 0; requestCapacity <= count; requestCapacity++)
+            {
+                Assert.Equal(currentCapacity, list.EnsureCapacity(requestCapacity));
+                Assert.Equal(currentCapacity, list.Capacity);
+            }
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(5)]
+        public void EnsureCapacity_CapacityIsAtLeastTheRequested(int count)
+        {
+            JCG.List<T> list = GenericListFactory(count);
+
+            int currentCapacity = list.Capacity;
+            int requestCapacity = currentCapacity + 1;
+            int newCapacity = list.EnsureCapacity(requestCapacity);
+            Assert.InRange(newCapacity, requestCapacity, int.MaxValue);
+        }
+
+        // J2N NOTE: Per comment in DoSetCapacity, "we consider reallocating the array a "modification" to the list to ensure sublists use the same array reference"
```

Comment:
Once we remove the version increment in `DoSetCapacity()` and fix the `CoModificationCheck()` we should be able to use the original BCL tests. See my comment above.